Protocol.md

```diff
@@ -181,17 +194,18 @@ and closing transactions.
    * [`point`:`funding_pubkey`]
    * [`spk`:`payout_spk`]
    * [`u64`:`payout_serial_id`]
-   * [`u16`:`num_funding_inputs`]
+   * [`bigsize`:`num_funding_inputs`]
```

Comment:
to make this change we need to either 

1. Add a version to the top of the `accept_dlc_tlv_v0`
2. revert this back to u16

The reason this is needed is to preserve backwards compatability with the previous version of the spec. This is how i've implemented the backwards compatible parse currently

```scala
    //parser for `offer_tlv`
    val fundingInputs = {
      if (protocolVersionOpt.isEmpty) {
        //means we are working with the old version of the spec
        iter.takeU16PrefixedList(() => iter.take(FundingInputV0TLV))
      } else {
        //means we are using version 1 of the spec
        iter.takeBigSizePrefixedList(() => iter.take(FundingInputV0TLV))
      }
    }
```

parser for the accept_tlv that is an attempt at backwards compatibility 

```scala
      //accept_tlv
      //
      //we don't have a version # here so we don't know what version we are trying to parse?
      //I suspect this code in the try block will not always cause an exception
      //so this is truely a guess.
      try {
        iter.takeU16PrefixedList(() => iter.take(FundingInputV0TLV))
      } catch {
        case scala.util.control.NonFatal(_) =>
          iter.takeBigSizePrefixedList(() => iter.take(FundingInputV0TLV))
      }
```

The crux of the issue is that if funding inputs are prefixed with a `u16` as required in the old version of the spec, a `bigsize` can be mistake as a `u16`. `bigsize` is required by the new version of the spec, which gives us ambiguity in the parser.

Reply:
I don't mind adding a version at the top of the message, but can't you just keep the version from the offer message?

Reply:
That doesn't work for a few reasons

1. Its a layer violation to require a database read just to parse a tlv correctly.
2. You cannot guarantee you will parse the `accept_tlv` since you don't know which serialization format it uses to obtain information to _make_ the database read. I guess you might be able to parse the first 32 bytes (`tempContractId`) to parse just the 32 bytes, make the database read, and then determine which format the offer had, but this is very cumbersome. 

Lets keep the messages -- at the binary serialization level -- context free so that you can parse the `offer`, `accept`, and `sign` tlvs independent of one another.

Reply:
Added a `protocol_version` field to the accept message and updated the test vectors. I wonder if it wouldn't be better to also add one to the sign message, what do you think?

Reply:
> I wonder if it wouldn't be better to also add one to the sign message, what do you think?

I think its probably a good idea. AFAIK, the `sign_tlv` does not have the same sort of ambiguity (`funding_inputs`) that `offer`,`accept` have so it isn't strictly necessary _now_, but it seems prudent to future proof a bit.

Reply:
> it seems prudent to future proof a bit.

Yeah I feel like that too. Updated docs and test vectors accordingly.